cargo install でビルドエラーが発生する

#  内容

以下のコマンドでインストールしようとするとエラーが発生する

```bash
$ cargo install --git https://github.com/cm-kazup0n/aurl-rust.git
```

```bash
error: cannot find derive macro `Parser` in this scope
 --> src/options.rs:5:10
  |
5 | #[derive(Parser, Debug)]
  |          ^^^^^^
  |
note: `Parser` is imported here, but it is only a trait, without a derive macro
 --> src/options.rs:2:25
  |
2 | use clap::{AppSettings, Parser};
  |                         ^^^^^^

error: cannot find attribute `clap` in this scope
 --> src/options.rs:6:3
  |
6 | #[clap(setting = AppSettings::ArgRequiredElseHelp)]
  |   ^^^^
  |
  = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
 --> src/options.rs:8:7
  |
8 |     #[clap(short, long, default_value = "default")]
  |       ^^^^
  |
  = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> src/options.rs:10:7
   |
10 |     #[clap(short = 'X', long, default_value = "GET")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> src/options.rs:13:7
   |
13 |     #[clap(short = 'H', long, multiple_values = true)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> src/options.rs:16:7
   |
16 |     #[clap(short, long)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> src/options.rs:18:7
   |
18 |     #[clap(long, default_value = "")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> src/options.rs:21:7
   |
21 |     #[clap(long, default_value = "none")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> src/options.rs:23:7
   |
23 |     #[clap(long, default_value = "30")]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

error: cannot find attribute `clap` in this scope
  --> src/options.rs:26:7
   |
26 |     #[clap(short = 'd', long)]
   |       ^^^^
   |
   = note: `clap` is in scope, but it is a crate, not an attribute

warning: unused import: `AppSettings`
 --> src/options.rs:2:12
  |
2 | use clap::{AppSettings, Parser};
  |            ^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0599]: no function or associated item named `parse` found for struct `Opts` in the current scope
  --> src/options.rs:31:11
   |
7  | pub struct Opts {
   | --------------- function or associated item `parse` not found for this
...
31 |     Opts::parse()
   |           ^^^^^ function or associated item not found in `Opts`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `parse`, perhaps you need to implement it:
           candidate #1: `Parser`

For more information about this error, try `rustc --explain E0599`.
warning: `aurl-rust` (bin "aurl-rust") generated 1 warning
error: failed to compile `aurl-rust v0.1.1 (https://github.com/cm-kazup0n/aurl-rust.git#83153b32)`, intermediate artifacts can be found at `/var/folders/8r/j4hys4253ng04f6wt4mjp4xm0000gp/T/cargo-install193iTI`
```

# 原因

Cargo.lock が無視されているそう。公式にIssueあり

https://github.com/rust-lang/cargo/issues/7169

# 回避策

以下のコマンドで回避できる

```bash
$ cargo install --git https://github.com/cm-kazup0n/aurl-rust.git --locked
```

# Close条件

- README に注意書きを記載？
- Clap のバージョンを上げて問題ないようにする？（問題がなくなるかはわからない）

--- a/README.md
+++ b/README.md
@@ -5,9 +5,58 @@ aurl-rust
 
 *aurl-rust* is yet another implementation of [aurl](https://github.com/classmethod/aurl)
 
-## Option
+## Install
 
-T.B.D
+Get the binary from [Releases](https://github.com/cm-kazup0n/aurl-rust/releases).
+
+or 
+
+```
+$ cargo install --git https://github.com/cm-kazup0n/aurl-rust.git --locked
+```
+
+installing with `--locked` is recommended.
+
+## Usage
+
+```
+aurl-rust 
+
+USAGE:
+    aurl-rust [OPTIONS] <URL>
+
+ARGS:
+    <URL>    
+
+OPTIONS:
+        --auth-header-template <AUTH_HEADER_TEMPLATE>
+            [default: ]
+
+    -d, --data <DATA>
+            
+
+    -h, --help
+            Print help information
+
+    -H, --header <HEADER>...
+            
+
+        --output <OUTPUT>
+            Output Option (case insensitive). curl: Output curl command snippet. none: Call URL with
+            Got AccessToken [default: none]
+
+    -p, --profile <PROFILE>
+            [default: default]
+
+        --timeout <TIMEOUT>
+            [default: 30]
+
+    -v, --verbose
+            A level of verbosity, and can be used multiple times
+
+    -X, --request <REQUEST>
+            [default: GET]
+```
 
 ## Profile
 

